libc/unistd_fsdir.c

```diff
@@ -58,12 +57,20 @@ TEST_SETUP(unistd_fsdir)
 
 TEST_TEAR_DOWN(unistd_fsdir)
 {
+	/* go back to the test working directory and assert it */
+	TEST_ASSERT_EQUAL_INT(0, chdir(testWorkDir));
+	TEST_ASSERT_NOT_NULL(getcwd(buf, sizeof(buf)));
+	TEST_ASSERT_EQUAL_STRING(testWorkDir, buf);
+
 	TEST_ASSERT_EQUAL_INT(0, remove(FNAME));
 }
 
 
 TEST(unistd_fsdir, getcwd)
 {
+	/* assumption that chdir("/") works fine when returning 0 */
```

Comment:
no assumption, You're verifying it below (using `getcwd`)

Reply:
Yeah, but I meant that we are testing `getcwd` and it could not work properly. Without such assumption we wouldn't know, whether it's because of problems with `gecwd` or `chdir`. The `chdir` function is tested after that, so I wanted to mention it.